miio/cooker_multi.py

```diff
@@ -0,0 +1,419 @@
+            return
+
+        max_minutes = (self.profile_bytes[10] * 60) + self.profile_bytes[11]
+        min_minutes = (self.profile_bytes[12] * 60) + self.profile_bytes[13]
+
+        if minutes < min_minutes or minutes > max_minutes:
+            return
+
+        self.profile_bytes[8] = math.floor(minutes / 60)
+        self.profile_bytes[9] = minutes % 60
+
+        self.update_checksum()
+
+    def is_schedule_enabled(self):
+        return (self.profile_bytes[14] & 0x80) == 0x80
+
+    def set_schedule_enabled(self, enabled):
+        if enabled:
+            self.profile_bytes[14] |= 0x80
+        else:
+            self.profile_bytes[14] &= 0x7F
+
+        self.update_checksum()
+
+    def set_schedule_duration(self, duration):
+        """Set the schedule time (delay before cooking) in minutes."""
+        schedule_flag = self.profile_bytes[14] & 0x80
+        self.profile_bytes[14] = math.floor(duration / 60) & 0xFF
+        self.profile_bytes[14] |= schedule_flag
+        self.profile_bytes[15] = (duration % 60 | self.profile_bytes[15] & 0x80) & 0xFF
+
+        self.update_checksum()
+
+    def is_akw_enabled(self):
+        return (self.profile_bytes[15] & 0x80) == 0x80
+
+    def set_akw_enabled(self, enabled):
+        if enabled:
+            self.profile_bytes[15] |= 0x80
+        else:
+            self.profile_bytes[15] &= 0x7F
+
+        self.update_checksum()
+
+    def calc_checksum(self):
+        crc = crcmod.mkCrcFun(0x11021, rev=False, initCrc=0x0, xorOut=0x0)(
+            self.profile_bytes
+        )
+        checksum = bytearray(2)
+        checksum[0] = (crc >> 8) & 0xFF
+        checksum[1] = crc & 0xFF
+        return checksum
+
+    def update_checksum(self):
+        self.checksum = self.calc_checksum()
+
+    def is_valid(self):
+        return len(self.profile_bytes) == 174 and self.checksum == self.calc_checksum()
+
+    def get_profile_hex(self):
+        return (self.profile_bytes + self.checksum).hex()
+
+
+class CookerStatus(DeviceStatus):
+    def __init__(self, data):
+        self.data = data
+
+    @property
+    def mode(self) -> OperationMode:
+        """Current operation mode."""
+        return OperationMode(self.data["status"])
+
+    @property
+    def menu(self) -> str:
+        """Selected menu id."""
+        try:
+            return COOKING_MENUS[self.data["menu"]]
+        except KeyError:
+            return "Unknown menu"
+
+    @property
+    def stage(self) -> str:
+        """Current stage if cooking."""
+        try:
+            return COOKING_STAGES[self.data["phase"]]["name"]
+        except KeyError:
+            return "Unknown stage"
+
+    @property
+    def temperature(self) -> Optional[int]:
+        """Current temperature, if idle.
+
+        Example values: 29
+        """
+        return self.data["temp"]
+
+    @property
+    def start_time(self) -> int:
+        """Start time of cooking?"""
+        return int(self.data["t_start"])
```

Comment:
If this is a timestamp, it'd be better to expose this as `datetime`?

Reply:
I removed this property.